oryx-app-serving/src/main/java/com/cloudera/oryx/app/serving/kmeans/model/KMeansServingModelManager.java

```diff
@@ -58,16 +58,14 @@ public void consume(Iterator<KeyMessage<String, String>> updateIterator) throws
       String message = km.getMessage();
       switch (key) {
         case "UP":
-          if (model == null) {
-            continue; // No model to interpret with yet, so skip it
+          if (model != null) {
```

Comment:
So this makes sense, although in other implementations where there is a lot of code that follows, I found it simpler to just take care of the `null` case upfront and leave it that way everywhere for consistency.